File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
from collections import defaultdict
def solution(name, yearning, photos):
    # 사진마다 추억 점수가 있다.
    # 사진의 추억 점수는 각 인물이 가지는 그리움 점수의 총 합
    # 각 인물을 key로 하여 그리움 점수를 value로 하기 없으면 0점 -> defaultdict
    # 각 사진을 순회 하며 각 인물의 점수를 모두 합산하여 저장하기
    
    # dict 정보 만들기
    dict = defaultdict(int)
    for n,y in zip(name,yearning):
        dict[n] = y
    #print(dict)
    
    # 순회하면서 합산하고 result에 저장하기
    result = []
    for photo in photos:
        sum = 0
        for n in photo:
            sum += dict[n]
        result.append(sum)
    #print(result)
    return result
    

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public static void log(Object o){
        System.out.println(o);
    }
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        /*
        각 사람의 점수를 map에 담기
        점수가 없는 경우에는 0
        */
        Map<String, Integer> score = new HashMap<>();
        int n = name.length;
        for(int i=0; i<n; i++){
            score.put(name[i], yearning[i]);
        }
        // for(String key: score.keySet()){
        //     log(key + " " + score.get(key));
        // }
        int[] result = new int[photo.length];
        for(int i=0; i<photo.length; i++){
            String[] people = photo[i];
            int sum = 0;
            for(String p: people){
                sum += score.getOrDefault(p, 0);
            }
            result[i] = sum;
        }
        return result;
    }
}